packages/opentelemetry/src/trace.ts

```diff
@@ -17,30 +16,18 @@ import { getCurrentHub } from './hub';
  *
  * If you want to create a span that is not set as active, use {@link startInactiveSpan}.
  *
- * Note that if you have not enabled tracing extensions via `addTracingExtensions`
- * or you didn't set `tracesSampleRate`, this function will not generate spans
- * and the `span` returned from the callback will be undefined.
+ * Note that you'll always get a span passed to the callback, it may just be a NonRecordingSpan if the span is not sampled.
  */
-export function startSpan<T>(spanContext: NodeExperimentalSpanContext, callback: (span: Span | undefined) => T): T {
+export function startSpan<T>(spanContext: OpenTelemetrySpanContext, callback: (span: Span) => T): T {
```

Comment:
Note: The API here has changed!! to what it was before.

Previously, the callback received `span: Span | undefined`, where `Span` was the span class from `@opentelemetry/sdk-trace-base`. This is not actually the same as you get from `tracer.startActiveSpan()`, and lead to some incompatibilities.

Instead, we now simply pass the proper span through, to keep this aligned. If the span is not sampled, you'll get a NonRecordingSpan (which still has all the write APIs but as noops). This also means users do not need to guard against `span` being undefined, which should simplify user code.